beamformer/unit_test/beamform_mult_kernel_test.py

```diff
@@ -61,72 +143,126 @@ def test_beamform_parametrised(batches, n_ants, num_channels, num_samples_per_ch
     # NOTE: test_id is a temporary inclusion meant to identify which complex multiply to call.
     test_id = "kernel"
 
-    n_channels_per_stream = num_channels // n_ants // 4
+    n_channels_per_stream = n_channels // n_ants // 4
     samples_per_block = 16
-    n_blocks = num_samples_per_channel // samples_per_block
-    coeff_gen = CoeffGenerator(batches, n_channels_per_stream, n_blocks, samples_per_block, n_ants)
+    n_blocks = n_samples_per_channel // samples_per_block
+    num_pols = 2
+
+    sample_period = 1 / 1712e6
+    num_delay_vals = n_channels_per_stream * n_beams * n_ants
+    delay_vals = []
+
+    for _ in range(num_delay_vals):
+        delay_vals.append(np.single((samples_delay) * sample_period))
+        delay_vals.append(np.single(0.0))
+        delay_vals.append(np.single(phase))
+        delay_vals.append(np.single(0.0))
+
+    # Change to numpy array and reshape
+    delay_vals = np.array(delay_vals)
+    delay_vals = delay_vals.reshape(n_channels_per_stream, n_beams, n_ants, 4)
 
     # 2. Initialise GPU kernels and buffers.
     ctx = accel.create_some_context(device_filter=lambda x: x.is_cuda, interactive=False)
     queue = ctx.create_command_queue()
 
-    # Create BeamformTemplate and link to buffer slots
-    beamform_mult_template = matrix_multiply.MatrixMultiplyTemplate(
-        ctx,
-        n_ants=n_ants,
-        n_channels=n_channels_per_stream,
-        n_samples_per_channel=num_samples_per_channel,
-        batches=batches,
-        test_id=test_id,
-    )
+    # Create compound
+    op_template = BeamformSeqTemplate(ctx, 
+        batches, 
+        num_pols, 
+        n_channels_per_stream, 
+        n_channels, 
+        n_blocks, 
+        samples_per_block, 
+        n_ants, 
+        n_beams, 
+        xeng_id, 
+        sample_period, 
+        n_samples_per_channel)
 
-    BeamformMult = beamform_mult_template.instantiate(queue, test_id)
-    BeamformMult.ensure_all_bound()
+    # Instantiate operational sequence
+    op = op_template.instantiate(queue)  
+    op.ensure_all_bound()
 
-    bufcoeff_device = BeamformMult.buffer("inCoeffs")
-    host_coeff = bufcoeff_device.empty_like()
+    # Create host buffers
+    buf_delay_vals_device = op.beamform_coeff.buffer("delay_vals")
+    host_delay_vals = buf_delay_vals_device.empty_like()
+    host_delay_vals = delay_vals
```

Comment:
This line doesn't do what you think it does. It clobbers the host buffer that you created in the previous line, and replaces it with a reference to `delay_vals`.

If you want to populate the contents of `host_delay_vals` with values from `delay_vals` then you'd do something like this:
```suggestion
    host_delay_vals[:] = delay_vals
```

It *shouldn't* be making a big deal, I certainly don't see how this could explain why you're getting your initial four values as wrong, but it could be a start.

Reply:
Thanks. Changed. The issue remains :(